### PR TITLE
generate test contents

### PR DESCRIPTION
**Pull request by lukegorman**
Add this https://github.com/British-Oceanographic-Data-Centre/COAsT/blob/develop/unit_testing/generate_unit_test_contents.py to GitActions

### CODE REVIEW

**Comment by thogar-computer** [comment]
@lukegorman @roje-bodc could be also add in a new git action diagram into the markdown file?

File: .github/workflows/unit_test_contents.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        run: |
+          git config --global user.name "ContentsBot"
+          git config --global author.name "ContentsBot"
+          git config --global user.email "bodcsoft@bodc.ac.uk"
+          git config --global author.email "bodcsoft@bodc.ac.uk"
+          git remote set-url origin https://x-access-token:${{ secrets.GITHUB_TOKEN }}@github.com/$GITHUB_REPOSITORY
+          git checkout $GITHUB_HEAD_REF
+          git diff-index --quiet $GITHUB_HEAD_REF || git commit -am "Commit generated unit test contents."
+          git push origin $GITHUB_HEAD_REF
+
```
Comment (MCazaly) [inline .github/workflows/unit_test_contents.yml:34]: Should we clear out these extra blank lines?

**Review by MCazaly — CHANGES_REQUESTED** [review]
Looking good. I agree with Tom's existing comments.

**Review by thogar-computer — APPROVED** [review]
Changes actioned and looking good

**Pull request merged**